src/libs/Notification/clearReportNotifications/index.native.ts

```diff
@@ -0,0 +1,38 @@
+import Airship, {PushPayload} from '@ua/react-native-airship';
+import Log from '@libs/Log';
+import {NotificationData} from '@libs/Notification/PushNotification/NotificationType';
+import CONST from '@src/CONST';
+import ClearReportNotifications from './types';
+
+function parseNotificationAndReportIDs(pushPayload: PushPayload) {
+    let payload = pushPayload.extras.payload;
+    if (typeof payload === 'string') {
+        payload = JSON.parse(payload);
+    }
+    const data = payload as NotificationData;
+    return {
+        notificationID: pushPayload.notificationId,
+        reportID: String(data.reportID),
```

Comment:
Why we need to cast to string here? Isn't the type on `ReportCommentNotificationData` wrong?

Reply:
The report ID i'm testing with gets parsed as a `number` so it might be wrong. It's also an old report though. I'll take a closer look at this

Reply:
Just checked and we are definitely sending `reportID` as `number` from the backend so the type is correct. We just need to cast as string here to make the comparison work